source/common/json/json_loader.h

```diff
@@ -4,7 +4,10 @@
 
 #include "common/common/non_copyable.h"
 
-struct json_t;
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wold-style-cast"
+#include "rapidjson/document.h"
```

Comment:
Do not leak this in header. Find a way around this even if you need to expose an interface.